include/deal.II/grid/reference_cell.h

```diff
@@ -183,6 +183,77 @@ namespace ReferenceCell
     return Point<dim>(+0.0, +0.0, +0.0);
   }
 
+  /**
+   * Return i-th unit tangential vector (array of dim-1 elements of Point<dim>)
+   * of a face of the reference cell. The vectors are arranged such that the
+   * cross product between the two vectors return the unit normal vector.
+   */
+  template <int dim>
+  inline Tensor<1, dim>
+  unit_tangential_vectors(const Type &       reference_cell,
+                          const unsigned int face_no,
+                          const unsigned int i)
+  {
+    if (reference_cell == get_hypercube(dim))
+      return GeometryInfo<dim>::unit_tangential_vectors[face_no][i];
+    else if (reference_cell == Type::Tri)
+      {
+        AssertDimension(i, 0);
+        static const std::array<Tensor<1, dim>, 3> table = {
+          {Point<dim>(1, 0),
```

Comment:
You don't need the detour of creating a `Point` object that is then cast to the base `Tensor` class:
```suggestion
          {{1, 0},
```

Reply:
same below